js/ez-navigation-hub.js

```diff
@@ -1,10 +1,135 @@
 (function () {
     /** @polymerElement */
-    class EzNavigationHub extends Polymer.Element {
+    class NavigationHub extends Polymer.Element {
         static get is() {
             return 'ez-navigation-hub';
         }
-    }
+        static get properties() {
+            return {
+                'activeZoneClass': {
+                    type: String,
+                    value: 'ez-active-zone',
+                },
+                'matchedLinkClass': {
+                    type: String,
+                    value: 'ez-matched-link',
+                },
+                'activeZone': {
+                    type: String,
+                    observer: '_updateActiveZone',
+                },
+                'matchedLinkUrl': {
+                    type: String,
+                    observer: '_highlightLink',
+                },
+            };
+        }
+
+
+        connectedCallback() {
+            super.connectedCallback();
+            this.addEventListener('click', (e) => {
+                if ( e.target.matches('[data-zone-identifier] a') ) {
+                    e.preventDefault();
+                    this.activeZone = e.target.parentNode.getAttribute('data-zone-identifier');
+                }
+            });
+        }
+
+        /**
+         * Highlight the active zone and update navigation by showing the related links.
+         * It's an observer of the 'activeZone' property.
+         *
+         * @param {String} newValue
+         * @param {String} oldValue
+         */
+        _updateActiveZone(newValue, oldValue) {
+            this._highlightZone(oldValue, newValue);
+            this._updateNavigation(oldValue, newValue);
+        }
+
+        /**
+         * Returns the zone corresponding to the given identifier.
+         *
+         * @param {String} identifier
+         * @return {HTMLElement}
+         */
+        _getZone(identifier) {
+            return this.querySelector('[data-zone-identifier="' + identifier + '"]');
+        }
+
+        /**
+         * Highlight the current active zone and unhighlight the previous one.
+         *
+         * @param {String} oldZoneIdentifier
+         * @param {String} newZoneIdentifier
+         */
+        _highlightZone(oldZoneIdentifier, newZoneIdentifier) {
+            if ( oldZoneIdentifier ) {
+                this._getZone(oldZoneIdentifier).classList.remove(this.activeZoneClass);
+            }
+            if ( newZoneIdentifier ) {
+                this._getZone(newZoneIdentifier).classList.add(this.activeZoneClass);
+            }
+        }
+
+        /**
+         * Returns every items corresponding to the given zone identifier.
+         *
+         * @param {String} identifier
+         * @return {Array|HTMLElement}
```

Comment:
It does not return an array nor an HTMLElement :)
See https://developer.mozilla.org/en-US/docs/Web/API/Element/querySelectorAll